internal/provider/data_source_packer_image.go

```diff
@@ -150,23 +151,18 @@ func dataSourcePackerImageRead(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
-	var channel *packermodels.HashicorpCloudPackerChannel
-
 	if channelSlug != "" {
-		channel, err = clients.GetPackerChannelBySlug(
+		channel, err := clients.GetPackerChannelBySlug(
 			ctx,
 			client,
 			loc,
 			bucketName,
 			channelSlug)
 		if err != nil {
 			return diag.FromErr(err)
+		} else if channel.Iteration == nil {
+			return diag.Errorf("Channel does not have an assigned iteration (channel: %s)", channelSlug)
```

Comment:
nit: I think this if check can stand on its own without the else statement. 
```suggestion
		} 
		if channel.Iteration == nil {
			return diag.Errorf("Channel does not have an assigned iteration (channel: %s)", channelSlug)
```